torch/csrc/DeviceAccelerator.cpp

```diff
@@ -18,8 +18,27 @@ void initModule(PyObject* module) {
 
   m.def("_accelerator_deviceCount", []() {
     auto device_type = at::accelerator::getAccelerator(false);
-    torch::utils::maybe_initialize_device(device_type);
-    return at::accelerator::deviceCount();
+    if (!device_type.has_value()) {
+      return static_cast<c10::DeviceIndex>(0);
+    }
+
+    // Why not call at::accelerator::deviceCount() directly like other
+    // accelerator python binding functions?
+    // 1. Some accelerators, such as CUDA, have a Python implementation of
+    // `device_count` that is non-poisoning.
```

Comment:
It assumes that all the accelerators expose `device_count` to Python. Otherwise, the implementation would throw an exception.

Reply:
Yes, we will follow the design pattern used in `torch.accelerator.is_available`.